crates/gosub_cairo/src/elements/text.rs

```diff
@@ -0,0 +1,233 @@
+        let lib = Library::init().expect("Failed to initialize FreeType");
+        let rc = Rc::new(ROBOTO_FONT.to_vec());
+        let default_face = lib.new_memory_face(rc, 0).expect("Failed to load font face");
+
+        // The FontContext struct holds the lib, ensuring it lives as long as all loaded faces
+        Mutex::new(GosubFontContext {
+            _library: lib,
+            font_ctx: FontContext::new(),
+            face_cache: HashMap::new(),
+            default_face,
+        })
+    });
+}
+
+#[allow(unused)]
+#[derive(Clone, Debug)]
+pub struct GsText {
+    // List of glyphs we need to show
+    glyphs: Vec<Glyph>,
+    // Actual utf-8 text (we don't have this yet)
+    text: String,
+    // Font we need to display (we need to have more info, like font familty, weight, etc.)
+    font: Font,
+    // Font size
+    fs: FP,
+    // List of coordinates for each glyph (?)
+    coords: Vec<NormalizedCoord>,
+    // Text decoration (strike-through, underline, etc.)
+    decoration: Decoration,
+}
+
+impl TText for GsText {
+    type Font = Font;
+
+    fn new<TL: TextLayout>(layout: &TL) -> Self
+    where
+        TL::Font: Into<Font>,
+    {
+        let font = layout.font().clone().into();
+        let fs = layout.font_size();
+
+        let glyphs = layout
+            .glyphs()
+            .iter()
+            .map(|g| Glyph {
+                id: g.id as GlyphID,
+                x: g.x,
+                y: g.y,
+            })
+            .collect();
+        let coords = layout.coords().iter().map(|c| NormalizedCoord::from_bits(*c)).collect();
+
+        Self {
+            glyphs,
+            text: String::new(),
+            font,
+            fs,
+            coords,
+            decoration: layout.decorations().clone(),
+        }
+    }
+}
+
+impl GsText {
+    pub(crate) fn render(obj: &RenderText<CairoBackend>, cr: cairo::Context) {
+        // let brush = &render.brush;
+        // let style: StyleRef = Fill::NonZero.into();
+        //
+        // let transform = render.transform.map(|t| t).unwrap_or(Transform::IDENTITY);
+        // let brush_transform = render.brush_transform.map(|t| t);
+
+        let base_x = obj.rect.x;
+        let base_y = obj.rect.y;
+        cr.move_to(base_x, base_y);
+
+        // Setup brush for rendering text
+        GsBrush::render(&obj.brush, cr.clone());
+
+        // This should be moved to the GosubFontContext::get_cairo_font_face(family: &str) method)
+        let font_face = unsafe {
+            LIB_FONT_FACE.with(|ctx_mutex| {
+                let mut ctx = ctx_mutex.lock().unwrap();
+
+                let ft_face = ctx.find_face_family("sans-serif");
+                let ft_face_ptr = ft_face.raw_mut() as *mut _ as *mut std::ffi::c_void;
+                let ff = cairo::ffi::cairo_ft_font_face_create_for_ft_face(ft_face_ptr, 0);
+                cairo::FontFace::from_raw_full(ff)
+            })
+        };
```

Comment:
I'm not 100% sure if this code is sound because you're locking the Mutex, creating a pointer in the FontFace, then dropping the Mutex lock, while still having the pointer to something in the Mutex.

I mean sure, this should always work, since it is a thread_local, so no one will move in the mutex for the next two lines.

I assume it should be fine, as long as cairo does not need the pointer to the font after the set_font_face call